Gems/Prefab/PrefabDependencyViewer/Code/Source/DirectedGraph.h

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/std/containers/unordered_map.h>
+#include <AzCore/std/containers/unordered_set.h>
+#include <AzToolsFramework/Prefab/Template/Template.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+#include <AzToolsFramework/Prefab/PrefabDomTypes.h>
+#include <AzCore/std/algorithm.h>
+#include <AzCore/std/containers/vector.h>
+#include <AzCore/std/containers/queue.h>
+#include <AzCore/std/containers/stack.h>
+
+namespace PrefabDependencyViewer::Utils
+{
+    class Node;
+    using NodeSet = AZStd::unordered_set<Node*>;
+    using ChildrenMap = AZStd::unordered_map<Node*, NodeSet>;
+    using TemplateId = AzToolsFramework::Prefab::TemplateId;
+
+    struct MetaData
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(MetaData, AZ::SystemAllocator, 0);
+
+        MetaData() = default;
+
+        MetaData(TemplateId tid, const char* source)
+            : m_tid(tid)
+            , m_source(source)
+        {
+        }
+
+        TemplateId GetTemplateId()
+        {
+            return m_tid;
+        }
+
+        const char* GetSource()
+        {
+            return m_source.c_str();
+        }
+
+    private:
+        TemplateId m_tid;
+        AZStd::string m_source;
+    };
+
+    class Node
+    {
+    public:
+        AZ_CLASS_ALLOCATOR(Node, AZ::SystemAllocator, 0);
+
+        Node(TemplateId tid, const char* source, Node* parent = nullptr)
+            : m_metaData(tid, source)
+            , m_parent(parent)
+        {
+        }
+
+        MetaData GetMetaData()
+        {
+            return m_metaData;
+        }
+
+        Node* GetParent()
+        {
+            return m_parent;
+        }
+
+        void SetParent(Node* parent)
+        {
+            m_parent = parent;
+        }
+
+    private:
+        MetaData m_metaData;
+        Node* m_parent;
+    };
+
+    class DirectedGraph
+    {
+    public:
+        DirectedGraph() {}
```

Comment:
Resolved. The reason I didn't use the default keyword in the first place was because {} was less words than default but they essentially are the same definitions so either is fine.

Reply:
I see. You are right. They are indeed subtle differences.